lib/SILOptimizer/SILCombiner/SILCombinerMiscVisitors.cpp

```diff
@@ -781,75 +782,137 @@ SILCombiner::visitInjectEnumAddrInst(InjectEnumAddrInst *IEAI) {
 
   // Ok, we have a payload enum, make sure that we have a store previous to
   // us...
-  SILBasicBlock::iterator II = IEAI->getIterator();
-  StoreInst *SI = nullptr;
-  InitEnumDataAddrInst *DataAddrInst = nullptr;
-  ApplyInst *AI = nullptr;
-  Operand *EnumInitOperand = nullptr;
-  for (;;) {
-    if (II == IEAI->getParent()->begin())
-      return nullptr;
-    --II;
-    SI = dyn_cast<StoreInst>(&*II);
-    if (SI) {
-      // Find a Store whose destination is taken from an init_enum_data_addr
-      // whose address is same allocation as our inject_enum_addr.
-      DataAddrInst = dyn_cast<InitEnumDataAddrInst>(SI->getDest());
-      if (DataAddrInst && DataAddrInst->getOperand() == IEAI->getOperand())
-        break;
-      SI = nullptr;
+  SILValue ASO = IEAI->getOperand();
+  llvm::SmallVector<InitEnumDataAddrInst *, 8> DataAddrVec;
+  llvm::SmallVector<InjectEnumAddrInst *, 8> InjectEnumAddrVec;
+  llvm::SmallPtrSet<SILInstruction *, 32> MayWriteSet;
+  for (auto usersIt : ASO->getUses()) {
```

Comment:
According to the LLVM style guide, we use caps in cases like this. I.e.:

for (auto Use : ASO->getUses()) { ... }